File: web/src/types/scrape.ts
export type ScrapeMethod = "http" | "llm_extract" | "fallback_text";

export type ScrapeResult = {
  url: string;
  job_title: string | null;
  company_name: string | null;
  location: string | null;
  job_description: string;
  raw_text: string;
  method: ScrapeMethod;
  success: boolean;
  error: string | null;
  char_count: number;
};

export type AutofillField = {
  key: string;
  label: string;
  value: string;
  category: string;
  multiline: boolean;
  hint: string;
};

export type AutofillResponse = {
  fields: AutofillField[];
  years_of_experience: number | null;
  primary_skills: string[];
};

File: src/jobassistant/schemas/scrape.py
from __future__ import annotations

from typing import Literal

from pydantic import BaseModel, HttpUrl, Field


ScrapeMethod = Literal["http", "llm_extract", "fallback_text"]


class ScrapeRequest(BaseModel):
    url: str = Field(..., min_length=1)


class ScrapeResult(BaseModel):
    url: str
    job_title: str | None = None
    company_name: str | None = None
    location: str | None = None
    job_description: str = ""
    raw_text: str = ""
    method: ScrapeMethod = "http"
    success: bool = True
    error: str | None = None
    char_count: int = 0
